test/org/sagebionetworks/bridge/dynamodb/DynamoAppConfigElementDaoTest.java

```diff
@@ -0,0 +1,281 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.sagebionetworks.bridge.TestConstants;
+import org.sagebionetworks.bridge.TestUtils;
+import org.sagebionetworks.bridge.exceptions.ConcurrentModificationException;
+import org.sagebionetworks.bridge.models.VersionHolder;
+import org.sagebionetworks.bridge.models.appconfig.AppConfigElement;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.datamodeling.PaginatedQueryList;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import com.amazonaws.services.dynamodbv2.model.ConditionalCheckFailedException;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DynamoAppConfigElementDaoTest {
+
+    private static final String ID_2 = "id2";
+
+    private static final String ID_1 = "id1";
+
+    @Mock
+    private DynamoDBMapper mockMapper;
+    
+    @Mock
+    private PaginatedQueryList<DynamoAppConfigElement> mockResults;
+    
+    @Captor
+    private ArgumentCaptor<DynamoDBQueryExpression<DynamoAppConfigElement>> queryCaptor;
+    
+    @Captor
+    private ArgumentCaptor<AppConfigElement> appConfigElementCaptor;
+    
+    private DynamoAppConfigElementDao dao;
+    
+    @Before
+    public void before() {
+        dao = new DynamoAppConfigElementDao();
+        dao.setMapper(mockMapper);
+    }
+    
+    @SuppressWarnings("unchecked")
+    @Test
+    public void getMostRecentElementsIncludeDeleted() {
+        when(mockMapper.batchLoad(any(List.class))).thenReturn(appConfigElementMapId1And2());
+        
+        List<AppConfigElement> returned = dao.getMostRecentElements(TestConstants.TEST_STUDY, true);
+        assertEquals(2, returned.size());
+        assertIdAndRevision(returned.get(0), ID_1, 3L);
+        assertIdAndRevision(returned.get(1), ID_2, 3L);
```

Comment:
Test should verify the DDB query. Also, we should probably return something "real" from the DDB query and pass it into the batchLoad() to verify that the data flow is working correctly.